pypeit/core/coadd.py

```diff
@@ -1390,179 +1557,135 @@ def ech_coadd(files,objids=None,extract='OPT',flux=True,giantcoadd=False,ordersc
-                ax2.plot(spec1d_final.wavelength, spec1d_final.flux, ls='steps-', color='b')
-
-                ax1.set_xlim([np.min(spec1d_final.wavelength.value), np.max(spec1d_final.wavelength.value)])
-                ax2.set_xlim([np.min(spec1d_final.wavelength.value), np.max(spec1d_final.wavelength.value)])
-                ax1.set_ylim(ylim)
-                ax2.set_ylim(ylim)
-                ax1.set_xlabel('Wavelength (Angstrom)')
-                ax1.set_ylabel('Flux')
-                ax2.set_ylabel('Flux')
-
-                plt.tight_layout(pad=0.2, h_pad=0., w_pad=0.2)
-
-                if len(qafile.split('.')) == 1:
-                    msgs.info("No fomat given for the qafile, save to PDF format.")
-                    qafile = qafile + '.pdf'
-                if qafile:
-                    plt.savefig(qafile)
-                    msgs.info("Wrote coadd QA: {:s}".format(qafile))
-                if debug:
-                    plt.show()
-                plt.close()
-
-            ### Do NOT remove this part althoug it is deprecated.
-            # we may need back to using this pieces of code after fixing the coadd_spectra problem on first order.
-            #kwargs['echelle'] = True
-            #kwargs['wave_grid_min'] = np.min(wave_grid)
-            #kwargs['wave_grid_max'] = np.max(wave_grid)
-            #spec1d_final = coadd_spectra(spectra_coadd_rebin, wave_grid_method=wave_grid_method, niter=niter,
-            #                                  scale_method=scale_method, do_offset=do_offset, sigrej_final=sigrej_final,
-            #                                  do_var_corr=do_var_corr, qafile=qafile, outfile=outfile,
-            #                                  do_cr=do_cr, debug=debug, **kwargs)
+            spec1d_final = merge_order(spectra_coadd_rebin, wave_grid, extract=extract, orderscale=orderscale,
+                                       niter=niter, sigrej_final=sigrej_final, SN_MIN_MEDSCALE=SN_MIN_MEDSCALE,
+                                       overlapfrac=overlapfrac, num_min_pixels=num_min_pixels,
+                                       phot_scale_dicts=phot_scale_dicts, qafile=qafile, outfile=outfile, debug=debug)
             return spec1d_final
         else:
             msgs.warn('Skipped merging orders')
             if outfile is not None:
                 for iord in range(len(spectra_list)):
-                    outfile_iord = outfile.replace('.fits','_ORDER{:04d}.fits'.format(iord))
-                    msgs.info('Saving the final calibrated spectrum of order {:d} as {:s}'.format(iord,outfile))
+                    outfile_iord = outfile.replace('.fits', '_ORDER{:04d}.fits'.format(iord))
+                    msgs.info('Saving the final calibrated spectrum of order {:d} as {:s}'.format(iord, outfile))
                     spectra_list[iord].write_to_fits(outfile_iord)
             return spectra_list
 
+    elif nfile==1:
```

Comment:
I would separate the cases of `nfiles == 1` and `nfiles > 1` into separate functions.